examples/multiplayer-netfox-state-machine/scripts/player.gd

```diff
@@ -0,0 +1,38 @@
+extends CharacterBody3D
+
+@export var state_machine: NetworkedStateMachine
+
+@onready var mesh_instance: MeshInstance3D = $MeshInstance3D
+@onready var current_state_label: Label3D = $StateNameLabel3D
+
+# Get the gravity from the project settings to be synced with RigidBody nodes.
+var gravity = ProjectSettings.get_setting("physics/3d/default_gravity")
+
+func _ready():
+	position = Vector3(0, 4, 0)
+	
+	# Wait a single frame, so player spawner has time to set input owner
+	await get_tree().process_frame
+	$RollbackSynchronizer.process_settings()
+	state_machine.set_state(&"Idle")
+	state_machine.state_changed.connect(func (_old_state, new_state):
+		if new_state:
+			current_state_label.text = new_state
```

Comment:
Shouldn't it be `new_state.name`, since these are `NetworkState` instances?

Reply:
good catch, I was emitting the wrong parameters too.